src/metabase/db/util.clj

```diff
@@ -13,10 +13,11 @@
   "Check if `model` is a toucan model.
   In toucan2 any keywords can be a model so it's always true for keyword."
   [model]
-  (if (keyword? model)
-    true
-    #_{:clj-kondo/ignore [:discouraged-var]}
-    (models/model? model)))
+  (or
```

Comment:
One thing you might want to think about here that is going to become more of a problem as you move away from requiring people to `(:require [metabase.models.card :refer [Card])` or stuff like that, is who ensures the stuff in `metabase.models.card` gets loaded. I think what you'll want to do here is put a `:before` method on `resolve-model` so if it sees a keyword like `:models/Card` it will ensure that `metabase.models.card` has been loaded before it tries to do stuff with it. I think putting this in `metabase.models.interface` (next to our other `resolve-model` method) is ok for now, of course eventually you'll have the problem of making sure **that** namespace gets loaded, but we can cross that bridge later.

I think you basically want something like this:

```clj
(ns imetabase.models.interface
  (:require 
   ...
   [camel-snake-kebab.internals.macros :as csk.macros]
   ...)
   
;;; define a custom CSK conversion function so we don't run into problems if the system locale is Turkish
(declare ^:private ->kebab-case-en) ; so Kondo doesn't complain
(csk.macros/defconversion "kebab-case-en" u/lower-case-en u/lower-case-en "-")

(methodical/defmethod t2.model/resolve-model :before :default
  [x]
  (when (and (keyword? x)
             (= (namespace x) "model")
             ;; don't try to require if it's already registered as a :metabase/model; this way ones defined in EE
             ;; namespaces won't break if they are loaded in some other way.
             (not (isa? x :metabase/model)))
    (let [model-namespace (str "metabase.models." (->kebab-case-en (name x)))]
      ;; use `classloader/require` which is thread-safe and plays nice with our plugins system
      (classloader/require model-namespace)))
  x)
```

Note the use of a custom [CSK ](https://github.com/clj-commons/camel-snake-kebab) conversion method there to use `u/lower-case-en`; `csk/->kebab-case` would be what we want to convert something like `DashboardCard` to `dashboard-card`, but it uses `str/lower-case`, which uses the system locale, and if the system locale is Turkish, we will have a bad time.

```clj
(metabase.test/with-locale "tr" (csk/->kebab-case "SoIsThisOk"))
;; => "so-ıs-this-ok"
```

Note the `ı`, not `i` like we'd expect. Don't need any more weird bugs like that.

Also we're skipping the `require` if the model is already registered, we do have some models defined with "non-standard" patterns (not in the corresponding `metabase.models.*` namespace), so this should still work as long as those namespaces get loaded some other way. Of course, it would be nicer if we could automatically load the correct namespace for every model instead of just erroring or something, but we can think about how to solve that problem. I think this is ok as a step 1

Reply:
that's a good catch but I wonder whether ensuring every toucan model is included in `metabase.models` is enough? 
I think I could write a test that ensures all toucan models have to put a `model.something/keepme` there.

And if it's not enough then why?

Reply:
Applied the change here: 40b6bb2e38. but would love to get the above question answered.